spring-boot/admin/src/main/java/eu/coatrack/admin/security/GatewayAuthFilter.java

```diff
@@ -44,22 +45,20 @@
  *
  * @author gr-hovest
  */
-public class GatewayAuthFilter extends GenericFilterBean {
+public class GatewayAuthFilter extends OncePerRequestFilter {
 
     private static final Logger log = LoggerFactory.getLogger(GatewayAuthFilter.class);
 
     @Override
-    public void doFilter(ServletRequest servletRequest, ServletResponse servletResponse, FilterChain filterChain) throws IOException, ServletException {
+    protected void doFilterInternal(HttpServletRequest servletRequest, HttpServletResponse servletResponse, FilterChain filterChain) throws ServletException, IOException {
 
-        if (servletRequest instanceof HttpServletRequest) {
-            final HttpServletRequest req = (HttpServletRequest) servletRequest;
-            final String keyParam = req.getParameter(Proxy.GATEWAY_API_KEY_REQUEST_PARAMETER_NAME);
+            final String keyParam = servletRequest.getParameter(Proxy.GATEWAY_API_KEY_REQUEST_PARAMETER_NAME);
             log.debug(String.format("request parameter '%s' has value '%s'", Proxy.GATEWAY_API_KEY_REQUEST_PARAMETER_NAME, keyParam));
 
             if (keyParam != null && !keyParam.isEmpty()) {
 
                 // Check credentials
-                HttpServletRequest httpRequest = (HttpServletRequest) servletRequest;
+                HttpServletRequest httpRequest = servletRequest;
```

Comment:
This is probably not required any more, right? Two variables referencing the same object?

Reply:
oh, that was a mistake.
Of course that is not required anymore, I will remove it

Reply:
Done